### ISSUE TITLE
Custom Route to Categories

### ISSUE THREAD

**Issue by keenojingco** [themosis/framework#345]
Hi,

More of a question than an issue.

I have a custom post type called blog-cs and I have defined a taxonomy for the categories and named it blog_categories for this as well. I'm trying to create a custom route that looks like this: site.com/blog/category/{category-slug}. I want to be able to use the slug to be able to query all the blog posts for that specific category.

On Laravel routes it's pretty straight forward. But I seem to be having trouble with this for Themosis routes. On a side note, I just started using your framework a few weeks ago and I'm slowly trying to understand it more and more.

Thanks

**Comment by dstollie** [comment]
Themosis uses WordPress' conditional tags for the WordPress routes. The following list shows all the keys with the conditional method it refers to:

``` php
    protected $conditions = [
        '404' => 'is_404',
        'archive' => 'is_archive',
        'attachment' => 'is_attachment',
        'author' => 'is_author',
        'category' => 'is_category',
        'date' => 'is_date',
        'day' => 'is_day',
        'front' => 'is_front_page',
        'home' => 'is_home',
        'month' => 'is_month',
        'page' => 'is_page',
        'paged' => 'is_paged',
        'postTypeArchive' => 'is_post_type_archive',
        'search' => 'is_search',
        'subpage' => 'themosis_is_subpage',
        'single' => 'is_single',
        'sticky' => 'is_sticky',
        'singular' => 'is_singular',
        'tag' => 'is_tag',
        'tax' => 'is_tax',
        'template' => 'is_page_template',
        'time' => 'is_time',
        'year' => 'is_year',
    ];
```

if /blog/category/{category-slug} is just the normall wordpress way to hit the overview page for a taxonomy you could just do:

``` php
Route::get('tax', ['my-taxonomy-slug', function() {
   return 'this is my category page'
}]
```

I think this should do.

Dennis

**Comment by jlambe** [comment]
This is a WordPress "issue", by default your custom taxonomy URI will use your taxonomy slug then the term slug like so: `http://domain.com/taxonomy-slug/term-slug/`

If you want to have a `blog` URI in front, you can define it either inside the admin permalinks page by setting a custom value like so for example: `blog/%postname%`. This will work but will add a `blog` uri in front of all your URIs.

Another way is to define a custom rewrite rule for your case by using the Rewrite API so you can create a custom taxonomy URI with a `blog` string in front.

**Comment by keenojingco** [comment]
@jlambe @dstollie Thanks both of you for your help and explanation. I got the route to work now. One more question. My url looks like this, `http://site.com/blog_categories/blog-all/`. As obviously this is what I have named my taxonomy slug, and term slug... 

Is it possible to beautify this without changing the slug? Let's say I wanted it to just display something like `http:://site.com/blog-category/all`
